plaso/parsers/wincc.py

```diff
@@ -0,0 +1,341 @@
+      disallowed_characters = ['\\', '/', ':', '*', '?', '"', '<', '>', '|']
+      for character in disallowed_characters:
+        if character in hostname:
+          error_string = (
+              'Hostname ({0!s}) can\'t contain the character {1:s} on line'
+              '{2:d}').format(hostname, character, line_number)
+          self._ParseValuesFail(
+              parser_mediator, first_line, exc, error_string)
+
+    event_data.hostname = hostname
+
+    event_data.source = values[7]
+
+    # The second to last field is the one that might contain unquoted separator.
+    # The last field can contain a string such as MSG_STATE_COME, MSG_STATE_GO,
+    # but also sometimes contains a message string.
+
+    text_message = ' '.join(values[8:])
+    event_data.message = text_message
+
+    parser_mediator.ProduceEventData(event_data)
+
+  def _ParseValuesFail(
+      self, parser_mediator, first_line, parent_exception, error_string):
+    if first_line:
+      raise errors.WrongParser(error_string) from parent_exception
+
+    parser_mediator.ProduceExtractionWarning(error_string)
+
+  def ParseFileObject(self, parser_mediator, file_object):
+    """Parses a bodyfile file-like object.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfVFS.
+      file_object (dfvfs.FileIO): file-like object.
+
+    Raises:
+      WrongParser: when the file cannot be parsed.
+    """
+    # Note that we cannot use the DSVParser here since this WinCC file format is
+    # not strict and clean file format.
+    line_reader = text_file.TextFile(
+        file_object, encoding=self.ENCODING, end_of_line='\r\n')
+
+    line_number = 0
+    first_line = True
+
+    # While the file is in UTF-16, the end_of_line character is '\r\n'.
+    # We manually strip this out.
+
+    try:
+      line = line_reader.readline().strip()
+    except UnicodeDecodeError as exception:
+      raise errors.WrongParser(
+          'unable to read line: {0:d} with error: {1!s}'.format(
+              line_number, exception))
+
+    while line:
+      if line.startswith('======>'):
+        # It seems that sometimes WinCC logs will end with a line starting like
+        # this, with the name of the next log file.
+        # But this is not always the case though.
+        line = line_reader.readline().strip()
+        continue
+      values = line.split(self.DELIMITER)
+      self._ParseValues(parser_mediator, line_number, values, first_line)
+
+      try:
+        line_number += 1
+        first_line = False
+        line = line_reader.readline().strip()
+      except UnicodeDecodeError as exception:
+        parser_mediator.ProduceExtractionWarning(
+            'unable to read line: {0:d} with error: {1!s}'.format(
+                line_number, exception))
+        break
+
+
+manager.ParsersManager.RegisterParser(WinCCSysLogParser)
```

Comment:
instead of the individual calls to RegisterParser use RegisterParsers([])

Reply:
done